rust/core-lib/src/file.rs

```diff
@@ -52,9 +52,9 @@ pub struct FileInfo {
 }
 
 pub enum FileError {
-    Io(io::Error),
-    UnknownEncoding,
-    HasChanged,
+    Io(io::Error, PathBuf),
```

Comment:
I tried a sample on the rust playground and it seems that path will not be displayed when `io::Error` is printed.
![capture](https://user-images.githubusercontent.com/7655600/41421676-4e4f9e2c-6fc5-11e8-8b1a-686ec4731958.PNG)
I also checked the [source file](https://doc.rust-lang.org/src/std/io/error.rs.html#66-68) of `io::Error` and found that path may not be tracked.
```rust
#[stable(feature = "rust1", since = "1.0.0")]
pub struct Error {
    repr: Repr,
}

enum Repr {
    Os(i32),
    Simple(ErrorKind),
    Custom(Box<Custom>),
}

#[derive(Debug)]
struct Custom {
    kind: ErrorKind,
    error: Box<error::Error+Send+Sync>,
}
```

Reply:
Cool, thanks for the explanation. :)

Reply:
This might warrant a comment in the code